libvast/vast/fbs/version.fbs

```diff
@@ -2,7 +2,12 @@ namespace vast.fbs;
 
 /// The global version of the persistent state. If any tables or contained data
 /// structures experience changes that are incompatible to a previous version,
-/// the version should get bumped.
+/// the version should get bumped. All version are put into the same enum so
+/// we can deduce the type of a flatbuffer from its version field, acting as
+/// an additional sanity check.
 enum Version : short {
-  v0,
+  Invalid,               // Default value
+  Segment_v0,
+  Index_v0,
+  Partition_v0,
 }
```

Comment:
Note to self: We discussed that we instead want to have two fields, `FlatbufferType` and `Version`.